builtin/common/settings/settingtypes.lua

```diff
@@ -69,9 +81,32 @@ local function parse_setting_line(settings, line, read_all, base_level, allow_se
 
 	-- category
 	local stars, category = line:match("^%[([%*]*)([^%]]+)%]$")
+	local category_context
```

Comment:
```suggestion
	local stars, category, category_context
	stars, category = line:match("^%[([%*]*)([^%]]+)%]$")
```
Would be more readable imho.

Reply:
Can't really agree, especially if also applied to the Lua pattern in the other place, it seems noisier as the names are listed three times instead of two times.